### PR TITLE
Secret Management Extension for AKV

### PR DESCRIPTION
**Pull request by BethanyZhou**
## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/KeyVault/KeyVault/Az.KeyVault.Extension/Az.KeyVault.Extension.psm1 [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+{
+    param (
+        [string] $SubscriptionId,
+        [string] $AzKVaultName
+    )
+
+    $azContext = Az.Accounts\Get-AzContext
+    if (($azContext -eq $null) -or ($azContext.Subscription.Id -ne $SubscriptionId))
+    {
+        throw "To use ${AzKVaultName} Azure vault, the current user must be logged into Azure account subscription ${SubscriptionId}. Run 'Connect-AzAccount -SubscriptionId ${SubscriptionId}'."
```
Comment (dingmeng-xue) [inline src/KeyVault/KeyVault/Az.KeyVault.Extension/Az.KeyVault.Extension.psm1:14]: Or `Set-AzContext -SubscriptionId ${SubscriptionId}`

File: src/KeyVault/KeyVault/Az.KeyVault.psd1 [hunk 2]
```diff
@@ -73,7 +73,7 @@ RequiredAssemblies = 'Microsoft.Azure.KeyVault.dll',
 FormatsToProcess = 'KeyVault.format.ps1xml'
 
 # Modules to import as nested modules of the module specified in RootModule/ModuleToProcess
-NestedModules = @('Microsoft.Azure.PowerShell.Cmdlets.KeyVault.dll')
+NestedModules = @('Microsoft.Azure.PowerShell.Cmdlets.KeyVault.dll', '.\Az.KeyVault.Extension')
```
Comment (dingmeng-xue) [inline src/KeyVault/KeyVault/Az.KeyVault.psd1:76]: Shall it be psm1 file?

**Review by dingmeng-xue — COMMENTED** [review]
2 minor comments.

**Comment by BethanyZhou** [comment]
Thanks @dingmeng-xue , I have updated codes as your suggestion.

**Review by dingmeng-xue — APPROVED** [review]

**Pull request merged**